src/main/java/com/google/sps/data/TripLocation.java

```diff
@@ -0,0 +1,29 @@
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * Inner value class for a location, with a Places ID
+ * (https://developers.google.com/places/place-id) string, weight representing
+ * the importance of the location, timestamp identifying the trip it's matched
+ * to, and owner email address string.
+ */
+@AutoValue
+public abstract class TripLocation {
+  public static final String ENTITY_PROPERTY_PLACE = "place_id";
+  public static final String ENTITY_PROPERTY_WEIGHT = "weight";
+  public static final String ENTITY_PROPERTY_TRIP = "trip";
+  public static final String ENTITY_PROPERTY_OWNER = "owner";
+
+  public static TripLocation create(String placeID, int weight, long timestamp, String owner) {
```

Comment:
Why isn't there a Builder as well for TripLocation?

Reply:
The rationale is that there are only 3 fields in the value class, so we don’t really need to chain calls with a builder. Using the create method is a one-liner here.

Reply:
I see, thank you